l10n_th_account_tax/models/account_partial_reconcile.py

```diff
@@ -28,11 +28,12 @@ def _create_tax_cash_basis_moves(self):
         if len(payment) == 1:
             self = self.with_context(payment_id=payment.id)
 
-        if (
-            self.debit_move_id.move_type == "in_refund"
-            and self.credit_move_id.move_type == "in_invoice"
-        ):
-            self = self.with_context(net_invoice_refund=1)
+        for move in self:
```

Comment:
```suggestion
        if all(move.debit_move_id.move_type == "in_refund" and move.credit_move_id.move_type == "in_invoice"  for move in self):
            self = self.with_context(net_invoice_refund=1)
```

For you code, If first move is in condition but second move is not in condition. self still context `net_invoice_refund`

I think we should check all self must in condition into add new context.

Reply:
Fixed 🙏